### PR TITLE
[Infra] Fullnode on LocalNet

### LINKED ISSUE #719: [Infrastructure] localnet config for full nodes
## Objective

Create a localnet configuration for full nodes, ensuring the ability to scale up and distinguish between validators and full nodes, and preventing the cluster manager from staking when increasing the number of full nodes.

## Origin Document

There is a need to deploy full nodes on the network, which does not act in the pacemaker lifecycle (i.e., voting on blocks) but only downloads and applies them. To enable this, a localnet configuration should be created for full nodes, allowing for easy scaling and differentiation between validators and full nodes. See the [full conversation here](https://discord.com/channels/824324475256438814/1006230632285868063/1100900789096038431).

## Goals

- Create and integrate a localnet configuration for full nodes in Tilt, enabling easy scaling that seamlessly integrates with the existing validator setup
- Implement a solution to clearly differentiate between validators and full nodes
- Prevent staking operations when increasing the number of full nodes

## Deliverable

- [ ] Implement a scalable localnet configuration for full nodes in Tilt
- [ ] Implement an environment variable or another appropriate method to distinguish between validator and full node roles within the network
- [ ] Modify the cluster manager to recognize and handle full nodes in addition to validators. Specifically, ensure that the cluster manager does not perform staking operations when scaling up the number of full nodes in the network

## Non-goals / Non-deliverables

- Ensuring full functionality of non-staked actors
- Implementing any consensus module changes related to full nodes

## General issue deliverables

- [ ] Update the appropriate CHANGELOG
- [ ] Update any relevant READMEs (local and/or global)
- [ ] Update any relevant global documentation & references
- [ ] If applicable, update the source code tree explanation
- [ ] If applicable, add or update a state, sequence or flowchart diagram using [mermaid](https://mermaid-js.github.io/mermaid/)

## Testing Methodology

- [ ] **All tests**: `make test_all`
- [ ] **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)

---

**Creator**: @jessicadaugherty

### PR DESCRIPTION
**Pull request by okdas**
## Description

Adds a fullnode (an unstaked validator) to the LocalNet

## Issue

Fixes #719 

## Type of change

Please mark the relevant option(s):

- [ ] New feature, functionality or library
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Major breaking change
- [ ] Documentation
- [ ] Other 

## List of changes

- Adds full node to LocalNet
- Makes helm chart agnostic of the protocol actor it runs
- Adds more private keys for other actors (except fisherman)

## Testing

- [x] `make develop_test`; if any code changes were made
- [ ] [Docker Compose LocalNet](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md); if any major functionality was changed or introduced
- [x] [k8s LocalNet](https://github.com/pokt-network/pocket/blob/main/build/localnet/README.md); if any infrastructure or configuration changes were made
- [x] remote DevNet

## Required Checklist

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added, or updated, [`godoc` format comments](https://go.dev/blog/godoc) on touched members (see: [tip.golang.org/doc/comment](https://tip.golang.org/doc/comment))
- [ ] I have tested my changes using the available tooling
- [ ] I have updated the corresponding CHANGELOG

### If Applicable Checklist

- [ ] I have updated the corresponding README(s); local and/or global
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added, or updated, [mermaid.js](https://mermaid-js.github.io) diagrams in the corresponding README(s)
- [ ] I have added, or updated, documentation and [mermaid.js](https://mermaid-js.github.io) diagrams in `shared/docs/*` if I updated `shared/*`README(s)

### CODE REVIEW

**Comment by okdas** [comment]
e2e passed on remote (sorry no CI for this yet)
image

File: build/debug_keybase/main.go [hunk 1]
```diff
@@ -143,24 +153,29 @@ func dumpKeybase(privateKeysYamlBytes []byte, targetFilePath string) {
...
-	}
-	validatorKeysMap := make(map[string]string)
+	decoder := yaml.NewDecoder(bytes.NewReader(privateKeysYamlBytes))
+	keysList := []string{}
+
+	for {
+		var secret K8sSecret
+
+		err := decoder.Decode(&secret)
+		if err != nil {
```
Comment (Olshansk) [inline build/debug_keybase/main.go:165]:
```suggestion
		if err := decoder.Decode(&secret); err != nil {
```

File: build/debug_keybase/main.go [hunk 2]
```diff
@@ -143,24 +153,29 @@ func dumpKeybase(privateKeysYamlBytes []byte, targetFilePath string) {
...
-		MetaData   map[string]string `yaml:"metadata"`
-		Type       string            `yaml:"type"`
-		StringData map[string]string `yaml:"stringData"`
-	}
-	if err := yaml.Unmarshal(privateKeysYamlBytes, &config); err != nil {
-		return nil, err
-	}
-	validatorKeysMap := make(map[string]string)
+	decoder := yaml.NewDecoder(bytes.NewReader(privateKeysYamlBytes))
+	keysList := []string{}
```
Comment (Olshansk) [inline build/debug_keybase/main.go:159]:
```suggestion
	keysList := make([]string, 0)
```

**Review by Olshansk — APPROVED** [review]
Some minor NITS + 1 question but nothing blocking so don't hesitate to merge in!

File: build/localnet/manifests/private-keys-misc.yaml [hunk 3]
```diff
@@ -0,0 +1,1007 @@
+apiVersion: v1
```
[the remark this answers is not in the document]
Comment (okdas) [reply build/localnet/manifests/private-keys-misc.yaml:1]: This file should have been deleted, thanks!

**Comment by Olshansk** [comment]
@okdas Following up on our offline conversation, let's fix this up and and get it over the finish line!

**Comment by okdas** [comment]
> @okdas Following up on our offline conversation, let's fix this up and and get it over the finish line!

Now it should be good to merge. :)

**Review by Olshansk — APPROVED** [review]

**Pull request merged**